### DISCUSSION TITLE
I'd really love the ability to save everything locally

### DISCUSSION THREAD

**Discussion by Yoric** [fariapp/fari-app#210] [Ideas]
As a Firefox Nightly user, I update Firefox every day *and* I cleanup my cookies, localStorage, indexedDB.
I'd really love the ability to save everything to a big JSON and reload it manually.

**Comment by RPDeshaies** [comment]
Hey!

Does firefox clean your localstorage automatically or do you do that manually ? just wondering out of curiosity.

I think this could go well with the idea of using Fari to generate campaigns and exporting them as a big blob to share publically on the web.

I'll add this to the backlog since I feel like there's other people who might find this useful

**Reply by Yoric** [reply]
> Does firefox clean your localstorage automatically or do you do that manually ? just wondering out of curiosity.

Nah, that's just my settings.

> I'll add this to the backlog since I feel like there's other people who might find this useful

Happy to contribute on this feature if someone can tell me where to start!

**Reply by RPDeshaies** [reply]
The project is written in Typescript + React.

The export import logic is located here: https://github.com/fariapp/fari/blob/3b4d2894aed52f8d8944856f5e9056d54bebce39/lib/domains/FariEntity/FariEntity.ts#L7-L13

and used in what I call the Character/Scene`Manager`: https://github.com/fariapp/fari/blob/0a282608d5dc5d552e080f409723ad4a245ed778/lib/components/CharactersManager/CharactersManager.tsx#L58-L90

I think that something cool we could do is a global `/export` route where you select everything you want to export using checkboxes or things like that and a global `/import` page that can import this file.

I separated the export mechanism in what I call "fariType" and this "global export" to me feels like a new entity type like `global` or something.

As a first version we could have a page with a big download button that exports everything and another page with a big import button which imports this type of file. (see `AppRouter.tsx` for creating pages)

Later down the line, we could give the ability to users to select which campaign they want to export or something.

**Comment by RPDeshaies** [comment]
I created a proper epic for this #222

**Comment by RPDeshaies** [comment]
Just an FYI that there's a  🚧   **BETA** 🚧  data manager available at https://fari.app/data

Be warned that it hasn't been fully teste, might contain bugs could corrupt your data.

I tested it, and it seemed fine but still wanted to write a warning message just in case.

**Reply by RPDeshaies** [reply]
@Yoric
